### PR TITLE
VRNIOP-3814: Adding Databus enhanced client

### PR DESCRIPTION
**Pull request by prasadinglearkin**

### CODE REVIEW

File: databus_client/queues/filters/metric_filter.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from databus_client.queues.filter_manager import FilterManager
+from databus_client.utils.common.databus_constants import DatabusMessageGroup
+
+
+class MetricFilter:
+
+    def pass_metric_filter(self, source=None, entry=None, message_group=None):
```
Comment (cnayakarkin) [inline databus_client/queues/filters/metric_filter.py:7]: convert to static. Other methods as well.

File: databus_client/queues/entity_queues/databus_metric_queue.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                    self.append_key_val_in_dict(message, ["status", "timestamp"], entry)
+                    self.logger.log(
+                        self.license_plate + ". Message received is identified as heartbeat. Bypassing filters")
+                    DatabusHeartBeatDbHandler.get_instance().add_to_queue(message)
+                else:
+                    """
+                    Getting filtered pass
+                    """
+                    self.logger.log(self.license_plate + "Passing through filter.")
+                    metric_filter = MetricFilter()
```
Comment (cnayakarkin) [inline databus_client/queues/entity_queues/databus_metric_queue.py:57]: import and call directly

**Review by cnayakarkin — COMMENTED** [review]
will continue.. please address comments till now

File: databus_client/queues/filter_manager.py [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+f = open(path)
+
+non_metric_message_group = ["applications", "vms", "flows", "hosts", "problems"]
+sub_metrics_message_group = ["vms-metrics", "hosts-metrics", "flows-metrics", "hosts-metrics", "nics-metrics",
+                             "switchports-metrics"]
+
+
+class FilterManager:
+
+    def set_filters(self, filter_dict=None):
```
Comment (cnayakarkin) [inline databus_client/queues/filter_manager.py:16]: convert to singleton and maintain one copy of lookup dict

**Review by cnayakarkin — CHANGES_REQUESTED** [review]
address previous comments

**Review by cnayakarkin — APPROVED** [review]
LGTM

File: databus_client/README.MD [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+Included SDK is compatible with vRNI 6.6, 6.7, 6.8 and onwards
+
+## Installing Python SDK [python 3.7]
+Checkout repo
+
+Then install the prerequisites with: pip install -r databus_client/requirements.txt
+
+Get the latest docker image from databus_client/dockerImages/
+
+Start docker container with the image
```
[the remark this answers is not in the document]
Comment (prasadinglearkin) [reply databus_client/README.MD]: Done

File: databus_client/endpoint/client_endpoint.py [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+            elif request_filter_dict["entity"] == 'vms':
+                result = f_manager.get_vms_filter(source=source)
+            elif request_filter_dict["entity"] == 'hosts':
+                result = f_manager.get_hosts_filter(source=source)
+            elif request_filter_dict["entity"] == 'flows':
+                result = f_manager.get_flows_filter(source=source)
+            elif request_filter_dict["entity"] == 'nics':
+                result = f_manager.get_nics_filter(source=source)
+            elif request_filter_dict["entity"] == 'problems':
+                result = f_manager.get_problems_filter(source=source)
```
[the remark this answers is not in the document]
Comment (prasadinglearkin) [reply databus_client/endpoint/client_endpoint.py]: Done

**Pull request merged**